libs/providers/flipt/src/lib/flipt-provider.ts

```diff
@@ -0,0 +1,228 @@
+          };
+        case 'UNKNOWN_EVALUATION_REASON':
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.UNKNOWN,
+          };
+        default:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.ERROR,
+          };
+      }
+    } catch (e) {
+      return {
+        value: defaultValue,
+        reason: StandardResolutionReasons.ERROR,
+        errorMessage: getErrorMessage(e),
+      };
+    }
+  }
+
+  async resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<string>> {
+    let value = await this.resolveFlagHelper(flagKey, 'string', defaultValue, context);
+    return value as ResolutionDetails<string>;
+  }
+
+  async resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<number>> {
+    let value = await this.resolveFlagHelper(flagKey, 'number', defaultValue, context);
+    return value as ResolutionDetails<number>;
+  }
+
+  async resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<U>> {
+    let value = await this.resolveFlagHelper(flagKey, 'json', defaultValue, context);
+    return value as ResolutionDetails<U>;
+  }
+
+  private async resolveFlagHelper<U extends JsonValue>(
+    flagKey: string,
+    flagType: PrimitiveTypeName,
+    defaultValue: PrimitiveType | U,
+    context: EvaluationContext,
+  ): Promise<ResolutionDetails<PrimitiveType | U>> {
+    const evalContext: Record<string, string> = transformContext(context);
+
+    try {
+      const variantEvaluation = await this.client?.evaluation.variant({
+        namespaceKey: this.namespace ?? 'default',
+        flagKey,
+        entityId: context.targetingKey ?? '',
+        context: {
+          ...evalContext,
+        },
+      });
+
+      if (variantEvaluation.reason === 'FLAG_DISABLED_EVALUATION_REASON') {
+        return {
+          value: defaultValue,
+          reason: StandardResolutionReasons.DISABLED,
+        };
+      }
+
+      if (!variantEvaluation.match) {
+        return {
+          value: defaultValue,
+          reason: StandardResolutionReasons.DEFAULT,
+        };
+      }
+
+      const flagValue: PrimitiveType | U = validateFlagType(
+        flagType,
+        flagType === 'json' ? variantEvaluation.variantAttachment : variantEvaluation.variantKey,
+      );
+
+      return {
+        value: flagValue,
+        reason: StandardResolutionReasons.TARGETING_MATCH,
+      };
+    } catch (e) {
+      if (e instanceof TypeMismatchError) {
+        throw e;
+      }
+
+      return {
+        value: defaultValue,
+        reason: StandardResolutionReasons.ERROR,
+        errorMessage: getErrorMessage(e),
+      };
+    }
```

Comment:
Please throw `GeneralError` so that OpenFeature will run registered error hooks.